### PR TITLE
Refactor ZipkinApi

### PR DESCRIPTION
**Pull request by pellared**
Partially addresses: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/32/files#r538755469

Changes proposed in this pull request:
- Remove direct reference to`TracerSettings` in `ZipkinApi` and other Zipkin related types
- Remove skip from a test which is working for me

Right now, the only setting is the URL to the endpoint. If we add any new Zipkin-specific setting then I think the `Tracer` constructor would have to be refactored to accept `IConfigurationSource` instead of `TracerSettings` so that we can move the parsing logic to more detailed components on the fly.

### CODE REVIEW

File: src/Datadog.Trace/Agent/ZipkinSerializer.cs [hunk 1]
```diff
@@ -16,15 +16,15 @@ internal class ZipkinSerializer
         // Don't serialize with BOM
         private readonly Encoding utf8 = new UTF8Encoding(false);
 
-        public void Serialize(Stream stream, Span[][] traces, TracerSettings settings)
```
Comment (pjanotti) [inline src/Datadog.Trace/Agent/ZipkinSerializer.cs:19]: Good removing this. This was a left-over from using some setting to set headers when doing Zipkin that was later removed but the `settings` were still being passed.

File: test/Datadog.Trace.Tests/CorrelationIdentifierTests.cs [hunk 2]
```diff
@@ -27,7 +27,7 @@ public void TraceIdSpanId_MatchActiveSpan()
             }
         }
 
-        [Fact(Skip = "This test is not compatible with the xUnit integration. Neither TraceId or SpanId are Zero.")]
```
Comment (pjanotti) [inline test/Datadog.Trace.Tests/CorrelationIdentifierTests.cs:30]: What did the history tell about this one?

**Review by pjanotti — APPROVED** [review]
A step in the right direction, eventually, we need to pass the configuration sources so we can implement each exporter supporting the OTel specification.

File: test/Datadog.Trace.Tests/CorrelationIdentifierTests.cs (hunk 2 above)
Comment (pellared) [reply test/Datadog.Trace.Tests/CorrelationIdentifierTests.cs:30]: The skip was introduced here: https://github.com/DataDog/dd-trace-dotnet/pull/802
I have not found any reason why we should skip it as it works 😉 
@tonyredondo is there something that I am missing?

File: test/Datadog.Trace.IntegrationTests/SendTracesToZipkinCollector.cs [hunk 3]
```diff
@@ -12,12 +12,8 @@ public class SendTracesToZipkinCollector
...
-            var settings = new TracerSettings()
-            {
-                AgentUri = new System.Uri("http://localhost:9411/api/v2/spans")
-            };
-
-            var api = new ZipkinApi(settings);
+            var agentUri = new System.Uri("http://localhost:9411/api/v2/spans");
+            var api = new ZipkinApi(agentUri);
             var agentWriter = new AgentWriter(api, statsd: null);
             _tracer = new Tracer(settings, agentWriter, sampler: null, scopeManager: null, statsd: null);
```
Comment (pjanotti) [inline test/Datadog.Trace.IntegrationTests/SendTracesToZipkinCollector.cs:18]: @pellared you still need the `settings` on this line.

**Review by pjanotti — CHANGES_REQUESTED** [review]
A build fix is required.

File: test/Datadog.Trace.IntegrationTests/SendTracesToZipkinCollector.cs (hunk 3 above)
Comment (pellared) [reply test/Datadog.Trace.IntegrationTests/SendTracesToZipkinCollector.cs]: My bad... Totally missed it 🤦

File: test/Datadog.Trace.Tests/CorrelationIdentifierTests.cs (hunk 2 above)
Comment (pellared) [reply test/Datadog.Trace.Tests/CorrelationIdentifierTests.cs]: It failed in Azure Pipelines. Yet, it really passed on my machine...

```
➜  Datadog.Trace.Tests git:(refactor-parsing-zipkin-settings) ✗ dotnet test

Passed!  - Failed:     0, Passed:   398, Skipped:     0, Total:   398, Duration: 22 s - /Users/rpajak/repos/opentelemetry-dotnet-instrumentation/test/Datadog.Trace.Tests/bin/Debug/netcoreapp2.1/Datadog.Trace.Tests.dll (netcoreapp2.1)

Passed!  - Failed:     0, Passed:   398, Skipped:     0, Total:   398, Duration: 22 s - /Users/rpajak/repos/opentelemetry-dotnet-instrumentation/test/Datadog.Trace.Tests/bin/Debug/netcoreapp3.0/Datadog.Trace.Tests.dll (netcoreapp3.0)

Passed!  - Failed:     0, Passed:   403, Skipped:     0, Total:   403, Duration: 26 s - /Users/rpajak/repos/opentelemetry-dotnet-instrumentation/test/Datadog.Trace.Tests/bin/Debug/netcoreapp3.1/Datadog.Trace.Tests.dll (netcoreapp3.1)
```

**Comment by pellared** [comment]
Looks like an unstable test failed:

```
 total of 1 test files matched the specified pattern.
[xUnit.net 00:00:03.0907007]     Datadog.Trace.Tests.CorrelationIdentifierTests.TraceIdSpanId_ZeroOutsideActiveSpan [SKIP]
  Skipped Datadog.Trace.Tests.CorrelationIdentifierTests.TraceIdSpanId_ZeroOutsideActiveSpan [1 ms]
[xUnit.net 00:00:07.4878646]     Datadog.Trace.Tests.SpanTests.Accurate_Duration(minimumSleepMilliseconds: 100) [FAIL]
  Failed Datadog.Trace.Tests.SpanTests.Accurate_Duration(minimumSleepMilliseconds: 100) [2 s]
  Error Message:
   Span duration difference (15.860259999999982ms) outside of allowed threshold 15ms. Expected average: 157.24365ms, actual average: 173.10390999999998ms
Expected: True
Actual:   False
  Stack Trace:
     at Datadog.Trace.Tests.SpanTests.Accurate_Duration(Int32 minimumSleepMilliseconds) in D:\a\1\s\test\Datadog.Trace.Tests\SpanTests.cs:line 134
  Standard Output Messages:
 [dd.trace_id=8997276081451242485,dd.span_id=169108037737575609]Average elapsed time: 157.2 ms
 [dd.trace_id=8997276081451242485,dd.span_id=169108037737575609]Average span time: 173.1 ms

```

Rerunning...

**Review by pjanotti — APPROVED** [review]

**Pull request merged**